packages/frontend/vite.config.ts

```diff
@@ -49,3 +41,24 @@ export default defineConfig({
         },
     },
 });
+
+/**
+ * Get common dependencies between frontend and ui-components packages.
+ * Needed to link other packages that use Solid.js:
+ * https://github.com/solidjs/solid/issues/1472
+ */
+function getCommonDependencies(): string[] {
+    const frontendPkg = JSON.parse(readFileSync(resolve(__dirname, "./package.json"), "utf-8"));
+    const uiComponentsPkg = JSON.parse(
+        readFileSync(resolve(__dirname, "../ui-components/package.json"), "utf-8"),
+    );
+
+    const frontendDeps = new Set(Object.keys(frontendPkg.dependencies || {}));
+    const uiComponentsDeps = new Set(Object.keys(uiComponentsPkg.dependencies || {}));
+
+    // @ts-expect-error: intersection method does exist on Set in our
+    // vite.config target i.e. NodeJS
+    const commonDeps = frontendDeps.intersection(uiComponentsDeps);
```

Comment:
Looks like this method is supported in Node v22+: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/intersection#browser_compatibility

We should bump the Node version used in CI to the latest LTS, which is apparently v24. Currently the CI is pegged to v20.